Add documentation for Gardener receiver (#61)
gardener/opentelemetry-collector: receiver/gardenerreceiver/docs/monitoring-a-landscape.md

# Monitoring a Gardener Landscape

This document describes how to deploy an OpenTelemetry Collector with the
Gardener receiver to monitor an entire Gardener landscape.

## Required Permissions

> The receiver only reads from the API server. The service-account used by the

Reviewer: ```suggestion
The receiver only reads from the API server. The `ServiceAccount` used by the
```

collector must be allowed to `get`, `list`, and `watch` the resources selected
via `resources`, plus the `SecretBinding` and `CredentialsBinding` resources
when `shoots` are enabled (these are needed to resolve billing-relevant
attributes on shoot info metrics).

API groups touched:

- `core.gardener.cloud` (`shoots`, `seeds`, `projects`, `secretbindings`)
- `seedmanagement.gardener.cloud` (`managedseeds`, `gardenlets`)
- `security.gardener.cloud` (`credentialsbindings`)

## Collector Configuration

To monitor a whole landscape, a minimal collector configuration is sufficient.

> The pipeline pairs a single `gardener` receiver with an `otlp` exporter that

Reviewer: ```suggestion
The pipeline pairs a single Gardener receiver with an `otlp` exporter that
```

Author: Keeping this since it is a reference to a configuration attribute

forwards the metrics to a downstream collector or backend.

> The receiver needs a `kubeconfig`, which points to the virtual garden API server.

Reviewer: ```suggestion
The receiver needs a kubeconfig, which points to the virtual garden API server.
```

Author: Keeping this since it is a reference to a configuration attribute
